pkgs/stdenv/adapters.nix

```diff
@@ -99,8 +99,12 @@ rec {
       mkDerivationFromStdenv = withOldMkDerivation old (stdenv: mkDerivationSuper: args:
       if stdenv.hostPlatform.isDarwin
       then throw "Cannot build fully static binaries on Darwin/macOS"
-      else (mkDerivationSuper args).overrideAttrs (args: {
-        NIX_CFLAGS_LINK = toString (args.NIX_CFLAGS_LINK or "") + " -static";
+      else (mkDerivationSuper args).overrideAttrs (args: if args ? env.NIX_CFLAGS_LINK then {
+        env = args.env // {
+          NIX_CFLAGS_LINK = toString args.env.NIX_CFLAGS_LINK + " -static";
+        };
+      } else {
+        NIX_CFLAGS_LINK = toString (args.env.NIX_CFLAGS_LINK or "") + " -static";
```

Comment:
Shouldn't this be `args.NIX_CFLAGS_LINK` here?

Reply:
hmm, that does indeed look wrong, I missed that in my review. Haven't actually managed to find anything in nixpkgs that builds statically and hits this case to verify tho.

Reply:
https://github.com/NixOS/nixpkgs/pull/332920